warp10/src/main/java/io/warp10/continuum/gts/MetadataSelectorMatcher.java

```diff
@@ -0,0 +1,190 @@
+    } catch (UnsupportedEncodingException uee) {
+      // Can't happen, we're using UTF-8
+    }
+
+    // build class selector pattern
+    if ("~.*".equals(classSelector)) {
+      this.classnamePattern = null;
+    } else if ("=".equals(classSelector) || "~".equals(classSelector)) {
+      this.classnamePattern = Pattern.compile(Pattern.quote("")).matcher("");
+    } else if (classSelector.startsWith("~")) {
+      this.classnamePattern = Pattern.compile(classSelector.substring(1)).matcher("");
+    } else if (classSelector.startsWith("=")) {
+      this.classnamePattern = Pattern.compile(Pattern.quote(classSelector.substring(1))).matcher("");
+    } else {
+      this.classnamePattern = Pattern.compile(Pattern.quote(classSelector)).matcher("");
+    }
+
+    // read label selector
+    String labelsSelection = m.group("labels");
+    Map<String, String> labelsSelectors = null;
+    try {
+      labelsSelectors = GTSHelper.parseLabelsSelectors(labelsSelection);
+    } catch (ParseException pe) {
+      throw new WarpScriptException(pe);
+    }
+
+    if (0 == labelsSelectors.size()) {
+      this.labelsPatterns = null;
+    } else {
+      this.labelsPatterns = new HashMap<>(labelsSelectors.size());
+      // build label patterns map
+      for (Entry<String, String> l: labelsSelectors.entrySet()) {
+        if (l.getValue().startsWith("=")) {
+          this.labelsPatterns.put(l.getKey(), Pattern.compile(Pattern.quote(l.getValue().substring(1))).matcher(""));
+        } else { //starts with ~ , otherwise Parse Exception in GTSHelper.parseLabelsSelectors
+          this.labelsPatterns.put(l.getKey(), Pattern.compile(l.getValue().substring(1)).matcher(""));
+        }
+      }
+    }
+
+    // read attribute selector, if any
+    Map<String, String> attributesSelectors = null;
+    String attributesSelection = m.group("attributes");
+    if (attributesSelection != null) {
+      try {
+        attributesSelectors = GTSHelper.parseLabelsSelectors(attributesSelection);
+      } catch (ParseException pe) {
+        throw new WarpScriptException(pe);
+      }
+      this.attributesPatterns = new HashMap<>(attributesSelectors.size());
+      // build label patterns map
+      for (Entry<String, String> l: attributesSelectors.entrySet()) {
+        if (l.getValue().startsWith("=")) {
+          this.attributesPatterns.put(l.getKey(), Pattern.compile(Pattern.quote(l.getValue().substring(1))).matcher(""));
+        } else { //starts with ~ , otherwise Parse Exception in GTSHelper.parseLabelsSelectors
+          this.attributesPatterns.put(l.getKey(), Pattern.compile(l.getValue().substring(1)).matcher(""));
+        }
+      }
+    } else {
+      this.attributesPatterns = null;
+    }
+  }
+
+  public boolean MetaDataMatch(Metadata metadata) {
+
+    // first, check classname
+    boolean classnameMatch = (null == this.classnamePattern) || null == metadata.getName()
```

Comment:
This means that if class is not set then a match will always occur?

Reply:
Yes. But it should not happen, as NEWGTS and NEWENCODER always set name to "".
Just a safety...

Reply:
given the method is public it should be assumed that it could be called with a Metadata instance whose name is not set, and in this case it should not match

Reply:
done, with earlier exit if classname does not match